### PR TITLE
ARROW-10908: [Rust][DataFusion] Update relevant tpch-queries with BETWEEN

### PR DESCRIPTION
**Pull request by seddonm1**
@Dandandan 
Fixes per https://github.com/apache/arrow/pull/8892#discussion_r541944358

### CODE REVIEW

**Review by Dandandan — APPROVED** [review]
LGTM, thanks!

**Review by jorgecarleitao — CHANGES_REQUESTED** [review]
Thanks a lot for this PR @seddonm1 .

`Between` is currently implemented as `<=` and `>=`, while the TPCH is implemented as `<` and `>`. They are semantically different (specially in types such as dates).

I do not know which one is right, but this introduces a (non-obvious) semantic change.

**Comment by Dandandan** [comment]
@jorgecarleitao I think it should have been using `>=` and `<=` before instead of being replaced in the benchmarks by `>` and `<`. So the change now fixes the behavior?

File: rust/benchmarks/src/bin/tpch.rs [hunk 1]
```diff
@@ -400,7 +400,7 @@ fn create_logical_plan(ctx: &mut ExecutionContext, query: usize) -> Result<Logic
                             (n1.n_name = 'FRANCE' and n2.n_name = 'GERMANY')
                             or (n1.n_name = 'GERMANY' and n2.n_name = 'FRANCE')
                         )
-                        and l_shipdate > date '1995-01-01' and l_shipdate < date '1996-12-31'
+                        and l_shipdate between date '1995-01-01' and date '1996-12-31'
```
Comment (alamb) [inline rust/benchmarks/src/bin/tpch.rs:403]: I think we can use the TPC-H Query validation section to resolve the use of `<` or `<=` etc by running the query against SF1 data and comparing the result

According to the spec, the expected answer is `123141078.23` at SF1 

http://www.tpc.org/tpc_documents_current_versions/pdf/tpc-h_v2.17.2.pdf
page 38


```
REVENUE
123141078.23
```

Comment (Dandandan) [reply rust/benchmarks/src/bin/tpch.rs:403]: That's a great idea, I agree this would be very valuable.

Comment (seddonm1) [reply rust/benchmarks/src/bin/tpch.rs:403]: I am working with Andy to increase the testing provided by the TPC-H queries here: https://github.com/apache/arrow/pull/8760#issuecomment-744076299

The plan is to verify results against the correct results when run in test mode.

Comment (seddonm1) [reply rust/benchmarks/src/bin/tpch.rs:403]: When implementing BETWEEN I went hunting for the ANSI SQL spec but the best I could do is find the Postgres implementation which confirms it is definitely INCLUSIVE.

https://www.postgresql.org/docs/13/functions-comparison.html

```
The BETWEEN predicate simplifies range tests:

a BETWEEN x AND y
is equivalent to

a >= x AND a <= y
```

**Review by jorgecarleitao — COMMENTED** [review]
Ok, I  went through the spec, and there is indeed no definition of `between`. As such, I am changing my review as neutral and will leave the experts take the decision.

**Review by jorgecarleitao — DISMISSED** [review]

**Review by alamb — APPROVED** [review]
I reviewed the TPCH spec to confirm this query is closer to the actual query text of the spec. 👍

**Comment by alamb** [comment]
I filed a separate subtask to track this work  https://issues.apache.org/jira/browse/ARROW-10817 as I think that is Arrow project standard (not to reuse tickets).

**Pull request closed without merging**